users/tests.py

```diff
@@ -137,10 +139,283 @@ def test_whoami(self):
+        response = self.client.post(url, data=json.dumps(data), content_type='application/json',
+                                    HTTP_AUTHORIZATION=fake_header)
+
+        self.assertEqual(response.content, response_content)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_password_update_unauthorized(self):
+        """Tests if it's not possible to update the current user password when unauthorized. """
+        url = reverse('password-update', kwargs={'version': 'v1'})
+        response = self.client.post(url, content_type='application/json')
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class UserProfileUpdateTest(BaseSingleUserTest):
+    """Tests the profile_update endpoint. """
+
+    def test_user_profile_update(self):
+        """Tests if it's possible to update the current user profile. """
+        url = reverse('token-obtain-pair', kwargs={'version': 'v1'})
+        auth = self.client.post(url, data=json.dumps(self._user), content_type='application/json')
+        fake_header = b'Bearer ' + json.loads(auth.content)['access'].encode()
+
+        data = {
+            'username': 'new_username',
+            'email': 'new_email@cusdeb.com',
+        }
+        url = reverse('user-profile-update', kwargs={'version': 'v1'})
+        response = self.client.post(url, data=json.dumps(data), content_type='application/json',
+                                    HTTP_AUTHORIZATION=fake_header)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_user_profile_update_with_missing_user_info(self):
+        """Tests if it's not possible to update the current user profile with missing user info. """
+        url = reverse('token-obtain-pair', kwargs={'version': 'v1'})
+        auth = self.client.post(url, data=json.dumps(self._user), content_type='application/json')
+        fake_header = b'Bearer ' + json.loads(auth.content)['access'].encode()
+
+        data = {'email': 'new_email@cusdeb.com'}
+        url = reverse('user-profile-update', kwargs={'version': 'v1'})
+        response = self.client.post(url, data=json.dumps(data), content_type='application/json',
+                                    HTTP_AUTHORIZATION=fake_header)
+
+        self.assertEqual(response.content, b'{"username": ["This field may not be blank."]}')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        data = {'username': 'new_username'}
+        url = reverse('user-profile-update', kwargs={'version': 'v1'})
+        response = self.client.post(url, data=json.dumps(data), content_type='application/json',
+                                    HTTP_AUTHORIZATION=fake_header)
+
+        self.assertEqual(response.content, b'{"email": ["This field may not be blank."]}')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_user_profile_updating_with_already_used_data(self):
```

Comment:
Let's rename it into `test_user_profile_update_using_existing_credentials` and update the docstring.